monai/auto3dseg/utils.py

```
@@ -330,34 +380,31 @@ def algo_from_pickle(pkl_filename: str, template_path: PathLike | None = None, *
 
     if len(template_paths_candidates) == 0:
-        # no template_path provided or needed
         algo = pickle.loads(algo_bytes)
         algo.template_path = None
     else:
         for i, p in enumerate(template_paths_candidates):
             try:
-                sys.path.append(p)
+                if p not in sys.path:
+                    sys.path.insert(0, p)
                 algo = pickle.loads(algo_bytes)
```

Ruff (0.14.13)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)